### PR TITLE
fastem-fixes-acq-calib-managers

### PR DESCRIPTION
**Pull request by Sabrina85**

### CODE REVIEW

File: src/odemis/acq/fastem.py
Comment (Sabrina85) [inline src/odemis/acq/fastem.py:232]: The time estimate is currently overridden by line 1621 in gui/cont/acquisition.py. I don't have a good idea how to handle this yet. The ROA, which we get the time estimate from, does not know about whether pre calibrations need to be run or not.

Comment (tepals) [reply src/odemis/acq/fastem.py:232]: You could add `pre_calibrate` as an attribute to the ROA class, and then update `roa.estimate_acquisition_time()` to add extra time when `pre_calibrate` is True. Then you could remove it as argument from `acquire` and `AcquisitionTask`, line 345 would then become `if self._roa.pre_calibrate:`

Comment (tepals) [reply src/odemis/acq/fastem.py:232]: you will then also need to update `test_pre_calibrate`

Comment (Sabrina85) [reply src/odemis/acq/fastem.py:232]: Yes that would make sense. I'll wait for the review from Éric whether I now have put setting the time in the right places and then I will change this.

File: src/odemis/acq/fastem.py
Comment (pieleric) [inline src/odemis/acq/fastem.py:454]: What's the point of initializing it to `None`? Is it so that even if get_config_asm() fails, the `finally` block doesn't fail? However, configure_asm() needs a dict properly field up, so passing None will fail anyway...

File: src/odemis/acq/fastem.py
Comment (pieleric) [reply src/odemis/acq/fastem.py:232]: That sounds like a good plan to me too.

**Review by pieleric — CHANGES_REQUESTED** [review]

File: src/odemis/acq/fastem.py
Comment (Sabrina85) [reply src/odemis/acq/fastem.py:454]: Correct. I have an extension for the `configure_asm()` case ready. I will push a new commit for this here.

Comment (Sabrina85) [reply src/odemis/acq/fastem.py:454]: Sorry, I will make a pr in the fastem-calibration repo for this not here.

**Review by tepals — CHANGES_REQUESTED** [review]
One small change needed

**Comment by Sabrina85** [comment]
> One small change needed

Which one? I don't see any comment?

File: src/odemis/acq/fastem.py
Comment (tepals) [inline src/odemis/acq/fastem.py:217]: Since you remove it here you also need to remove it where this function is called, in fastem_acq on line 461.

**Comment by tepals** [comment]
> > One small change needed
> 
> Which one? I don't see any comment?

it disappeared, I added it again

File: src/odemis/acq/fastem.py
Comment (Sabrina85) [reply src/odemis/acq/fastem.py:217]: Ah that got added during the rebase again 🙈

**Review by tepals — APPROVED** [review]

**Review by pieleric — APPROVED** [review]

**Pull request merged**
